pkg/osutil/interrupt_unix.go

```diff
@@ -60,9 +60,7 @@ func HandleInterrupts(lg *zap.Logger) {
 
 		interruptExitMu.Lock()
 
-		if lg != nil {
-			lg.Info("received signal; shutting down", zap.String("signal", sig.String()))
-		}
+		lg.Info("received signal; shutting down", zap.String("signal", sig.String()))
```

Comment:
Please add an assertion something like example below at the beginning of this function.

https://github.com/etcd-io/etcd/blob/4d57eb8d3b2190bfdba3c65f8eb93c0349fc6dcc/pkg/schedule/schedule.go#L92

Reply:
https://github.com/etcd-io/etcd/pull/14422#discussion_r962517555 
Actually it's would not be nil. `lg != nil` should be historical reasons. 
But I am ok add this.

Reply:
Please note that not only etcd may use this function, other applications may also use this function. We should clearly state the the `lg` shouldn't be nil using assertion.

Reply:
That make sense. DONE！